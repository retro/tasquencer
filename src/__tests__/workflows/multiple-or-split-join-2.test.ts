import { Effect } from 'effect';
import { expect, it } from 'vitest';

import { Builder, IdGenerator, Service } from '../../index.js';
import { getEnabledTaskNames, makeIdGenerator } from '../shared.js';

const workflowDefinition = Builder.workflow<{ isBToCEnabled: boolean }>()
  .withName('multiple-or-join-2')
  .startCondition('start')
  .task('A', Builder.emptyTask().withSplitType('and'))
  .task('B', Builder.emptyTask().withSplitType('xor'))
  .task('C', Builder.emptyTask().withJoinType('or'))
  .task('D', Builder.emptyTask().withJoinType('or'))
  .endCondition('end')
  .connectCondition('start', (to) => to.task('A'))
  .connectTask('A', (to) => to.task('B').task('C'))
  .connectTask('B', (to) =>
    to
      .task('C', ({ context }) => Effect.succeed(context.isBToCEnabled))
      .defaultTask('D')
  )
  .connectTask('C', (to) => to.task('D'))
  .connectTask('D', (to) => to.condition('end'));

it('runs a net with multiple "or" splits and "or" joins (1)', () => {
  const program = Effect.gen(function* ($) {
    const idGenerator = makeIdGenerator();

    const service = yield* $(
      workflowDefinition.build(),
      Effect.flatMap((workflow) =>
        Service.initialize(workflow, {
          isBToCEnabled: true,
        })
      ),
      Effect.provideService(IdGenerator, idGenerator)
    );

    expect(yield* $(service.getState())).toMatchSnapshot();

    yield* $(service.start());
    const state1 = yield* $(service.getState());
    expect(state1).toMatchSnapshot();
    expect(getEnabledTaskNames(state1)).toEqual(new Set(['A']));

    yield* $(service.startTask('A'));
    const state2 = yield* $(service.getState());
    expect(state2).toMatchSnapshot();
    expect(getEnabledTaskNames(state2)).toEqual(new Set(['B']));

    yield* $(service.startTask('B'));
    const state3 = yield* $(service.getState());
    expect(state3).toMatchSnapshot();
    expect(getEnabledTaskNames(state3)).toEqual(new Set(['C']));

    yield* $(service.startTask('C'));
    const state4 = yield* $(service.getState());
    expect(state4).toMatchSnapshot();
    expect(getEnabledTaskNames(state4)).toEqual(new Set(['D']));

    yield* $(service.startTask('D'));
    const state5 = yield* $(service.getState());
    expect(state5).toMatchSnapshot();
    expect(state5.workflows[0]?.state).toBe('completed');
  });

  Effect.runSync(program);
});

it('runs a net with multiple "or" splits and "or" joins (2)', () => {
  const program = Effect.gen(function* ($) {
    const idGenerator = makeIdGenerator();

    const service = yield* $(
      workflowDefinition.build(),
      Effect.flatMap((workflow) =>
        Service.initialize(workflow, {
          isBToCEnabled: false,
        })
      ),
      Effect.provideService(IdGenerator, idGenerator)
    );

    expect(yield* $(service.getState())).toMatchSnapshot();

    yield* $(service.start());
    const state1 = yield* $(service.getState());
    expect(state1).toMatchSnapshot();
    expect(getEnabledTaskNames(state1)).toEqual(new Set(['A']));

    yield* $(service.startTask('A'));
    const state2 = yield* $(service.getState());
    expect(state2).toMatchSnapshot();
    expect(getEnabledTaskNames(state2)).toEqual(new Set(['B']));

    yield* $(service.startTask('B'));
    const state3 = yield* $(service.getState());
    expect(state3).toMatchSnapshot();
    expect(getEnabledTaskNames(state3)).toEqual(new Set(['C']));

    yield* $(service.startTask('C'));
    const state4 = yield* $(service.getState());
    expect(state4).toMatchSnapshot();
    expect(getEnabledTaskNames(state4)).toEqual(new Set(['D']));

    yield* $(service.startTask('D'));
    const state5 = yield* $(service.getState());
    expect(state5).toMatchSnapshot();
    expect(state5.workflows[0]?.state).toBe('completed');
  });

  Effect.runSync(program);
});
