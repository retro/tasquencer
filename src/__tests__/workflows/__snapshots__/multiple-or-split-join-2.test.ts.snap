// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`runs a net with multiple "or" splits and "or" joins (1) 1`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 0,
      "name": "A",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "B",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "C",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": true,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "initialized",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (1) 2`] = `
{
  "conditions": [
    {
      "marking": 1,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 0,
      "name": "A",
      "state": "enabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "B",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "C",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": true,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "started",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (1) 3`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 1,
      "name": "A",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "B",
      "state": "enabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "C",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": true,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "started",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (1) 4`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 1,
      "name": "A",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "B",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "C",
      "state": "enabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": true,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "started",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (1) 5`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 1,
      "name": "A",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "B",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "C",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "enabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": true,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "started",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (1) 6`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 1,
      "name": "A",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "B",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "C",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "D",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": true,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "completed",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (2) 1`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 0,
      "name": "A",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "B",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "C",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": false,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "initialized",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (2) 2`] = `
{
  "conditions": [
    {
      "marking": 1,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 0,
      "name": "A",
      "state": "enabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "B",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "C",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": false,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "started",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (2) 3`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 1,
      "name": "A",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "B",
      "state": "enabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "C",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": false,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "started",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (2) 4`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 1,
      "name": "A",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "B",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "C",
      "state": "enabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "disabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": false,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "started",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (2) 5`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 1,
      "name": "A",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "B",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "C",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 0,
      "name": "D",
      "state": "enabled",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": false,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "started",
    },
  ],
}
`;

exports[`runs a net with multiple "or" splits and "or" joins (2) 6`] = `
{
  "conditions": [
    {
      "marking": 0,
      "name": "start",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 1,
      "name": "end",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->B",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:A->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->C",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:B->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "marking": 0,
      "name": "implicit:C->D",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "tasks": [
    {
      "generation": 1,
      "name": "A",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "B",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "C",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
    {
      "generation": 1,
      "name": "D",
      "state": "completed",
      "workflowId": "workflow-1",
      "workflowName": "multiple-or-join-2",
    },
  ],
  "workItems": [],
  "workflows": [
    {
      "context": {
        "isBToCEnabled": false,
      },
      "id": "workflow-1",
      "name": "multiple-or-join-2",
      "parent": null,
      "state": "completed",
    },
  ],
}
`;
